### PR TITLE
[Ignore][SERVICE-453] Fix random test failures around channel-changed events

### PR DESCRIPTION
**Pull request by bryangaleopenfin**
Note this current attempt at a solution does not fix the problem - https://appoji.jira.com/browse/SERVICE-453

### CODE REVIEW

File: src/provider/FDC3.ts [hunk 1]
```diff
@@ -68,7 +68,10 @@ export class FDC3 {
         const requestedApp: Application | undefined = applications.find((app: Application) => app.name === payload.name);
         return new Promise<void>((resolve: () => void, reject: (reason: Error) => void) => {
             if (requestedApp) {
-                this.openApplication(requestedApp, payload.context).then(resolve, reject);
+                this.openApplication(requestedApp, payload.context).then(() => {
+                    // Delay so that we can be the app has joined the global channel before returning
```
Comment (MichaelMCoates) [inline src/provider/FDC3.ts:72]: `we can be the app`

Comment (bryangaleopenfin) [reply src/provider/FDC3.ts:72]: Meant to be 'we can be sure the app'! 👍

File: src/provider/APIHandler.ts [hunk 2]
```diff
@@ -65,9 +65,9 @@ export class APIHandler<A extends string, P extends {[K in A]: unknown}, R exten
 
         // The 'onConnection' callback fires *just before* the channel is ready.
         // Delaying the firing of our signal slightly, to ensure client is definitely contactable.
-        setImmediate(() => {
+        setTimeout(() => {
             this.onConnection.emit(app);
-        });
+        }, 100);
```
Comment (tomrobertsOF) [inline src/provider/APIHandler.ts:70]: Is 100 the lowest number that will work here? I would prefer to have as low of a delay as we can get away with.

File: test/demo/contextChannels.test.ts [hunk 3]
```diff
@@ -2,6 +2,7 @@ import 'jest';
 import {connect, Fin, Identity, Application} from 'hadouken-js-adapter';
 
 import * as fdc3Remote from './utils/fdc3RemoteExecution';
+import {delay} from './utils/delay';
```
Comment (tomrobertsOF) [inline test/demo/contextChannels.test.ts:5]: You import but don't use delay.

File: test/demo/contextListener.test.ts [hunk 4]
```diff
@@ -3,6 +3,7 @@ import {OrganizationContext} from '../../src/client/main';
 
 import {fin} from './utils/fin';
 import * as fdc3Remote from './utils/fdc3RemoteExecution';
+import {delay} from './utils/delay';
```
Comment (tomrobertsOF) [inline test/demo/contextListener.test.ts:6]: as above, imported but not used.

File: test/demo/contextChannels.test.ts (hunk 3 above)
Comment (tomrobertsOF) [reply test/demo/contextChannels.test.ts:5]: (unless I missed it somewhere)

**Pull request closed without merging**